pkg/matching/compiler.go

```diff
@@ -0,0 +1,44 @@
+package matching
+
+import (
+	"sync"
+
+	"github.com/cespare/xxhash"
+	"github.com/elastic/go-freelru"
+	"github.com/kyverno/kyverno-json/pkg/core/assertion"
+	"github.com/kyverno/kyverno-json/pkg/core/templating"
+)
+
+type Compiler struct {
+	templating.Compiler
+	*freelru.SyncedLRU[string, func() (assertion.Assertion, error)]
+}
+
+func hashStringXXHASH(s string) uint32 {
+	sum := xxhash.Sum64String(s)
+	return uint32(sum) //nolint:gosec
+}
```

Comment:
The cache library uses `github.com/cespare/xxhash/v2` for hashing: https://github.com/elastic/go-freelru/blob/a8ff11dc7ea0d398a6b9aefe62aeff605f2d8962/bench/hash.go#L191